packages/components/text/src/Text.tsx

```diff
@@ -126,7 +146,8 @@ export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTo
       ...extra,
       onPress,
       numberOfLines: truncate || !wrap ? 1 : 0,
-      style: mergeStyles(tokenStyle, props.style, extra?.style),
+      style: mergeStyles(tokenStyle, props.style, extra?.style, scaleStyleAdjustments),
+      ...(dynamicTypeVariant !== undefined && { allowFontScaling: false }), // TODO(#2268): Remove once RN Core properly supports Dynamic Type scaling
```

Comment:
For consistency sake, could we put the spread next to the other spreads? 
I also would have expected `...rest` to be the last line so that everything is user-overridable, but I guess that's outside this PR's scope.